api/src/functionalTest/java/uk/gov/hmcts/payment/functional/dsl/PaymentsV2TestDsl.java

```diff
@@ -0,0 +1,190 @@
+package uk.gov.hmcts.payment.functional.dsl;
+
+import io.restassured.RestAssured;
+import io.restassured.http.ContentType;
+import io.restassured.parsing.Parser;
+import io.restassured.response.Response;
+import io.restassured.specification.RequestSpecification;
+import org.apache.commons.lang.RandomStringUtils;
+import org.assertj.core.api.Assertions;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.payment.api.contract.CardPaymentRequest;
+import uk.gov.hmcts.payment.api.contract.PaymentDto;
+import uk.gov.hmcts.payment.api.v1.contract.PaymentOldDto;
+import uk.gov.hmcts.payment.api.v1.contract.RefundPaymentRequestDto.RefundPaymentRequestDtoBuilder;
+import uk.gov.hmcts.payment.functional.tokens.ServiceTokenFactory;
+import uk.gov.hmcts.payment.functional.tokens.UserTokenFactory;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+import java.util.function.Consumer;
+
+@Component
+@Scope("prototype")
+public class PaymentsV2TestDsl {
+    private final Map<String, String> headers = new HashMap<>();
+
+    @Value("${test.url:http://localhost:8080}")
```

Comment:
will that work? or do we need to pick it up from properties? default url is fine. not sure if this port will be available everywhere